src/ExceptionMechanism.php

```diff
@@ -26,18 +26,28 @@ final class ExceptionMechanism
      */
     private $handled;
 
+    /**
+     * @var array|null Arbitrary extra data that might help the user understand
+     *                 the error thrown by this mechanism
+     */
+    private $data;
+
     /**
      * Class constructor.
      *
-     * @param string $type    Unique identifier of this mechanism determining
-     *                        rendering and processing of the mechanism data
-     * @param bool   $handled Flag indicating whether the exception has been
-     *                        handled by the user (e.g. via try..catch)
+     * @param string     $type    Unique identifier of this mechanism determining
+     *                            rendering and processing of the mechanism data
+     * @param bool       $handled Flag indicating whether the exception has been
+     *                            handled by the user (e.g. via try..catch)
+     * @param array|null $data    Arbitrary extra data that might help the user
+     *                            understand the error thrown by this mechanism
+     * @psalm-param array<string, mixed>|null $data
```

Comment:
We can directly use `@param array<string, mixed>` instead of adding `@psalm-param`, as we did it already in a few other places of the codebase